### ISSUE TITLE
More improvements

### ISSUE THREAD

**Issue by xoxys** [restic/calens#9]
Hi, I would like to work on some more improvements:

- switch to `urfave/cli` to easily work with multiple config sources; I'm mostly interested in setting options as env vars
- add CI for
  - test automation
  - publish pre-build binaries with GH release
  - publish a Docker image
- documentation

I completely understand if you want to leave the base as it is, but I wanted to ask you for your opinion instead of just hacking around in a fork :)

**Comment by fd0** [comment]
Hm, I prefer `cobra`, which we also use in restic, so we don't have to use two different frameworks. You can extract configuration options via environment variables like here:

https://github.com/restic/restic/blob/0ae02f30303aecac26f8b5cb54f492d240c6cde7/cmd/restic/global.go#L104

CI for test automation is great, I'm not so sure about the docker images (yet at least).

Documentation would be great :)

**Comment by xoxys** [comment]
Thanks for your feedback! Cobra is fine, I'll look into the env var usage. For CI, I guess you would also like to use existing CI Appveyor? Not sure if I could help with that never worked with it :)

**Comment by fd0** [comment]
Nah, I'd love to find the time to migrate restic to GitHub Actions, so I'd prefer that for calens, too. Here are some workflows to get you started (for a different project of mine): https://github.com/happal/taifun/tree/master/.github/workflows
